tensorboard/plugins/profile/pod_viewer/details_card/details-card.ts

```diff
@@ -0,0 +1,160 @@
+
+Polymer({
+  is: 'details-card',
+  properties: {
+    nodes: {
+      type: Array,
+      notify: true,
+      observer: 'updateCard_',
+    },
+    name: {
+      type: String,
+      value: null,
+    },
+    id: {
+      type: Number,
+    },
+    utilization: {
+      type: Number,
+    },
+    isChannel: {
+      type: Boolean,
+      value: false,
+    },
+    isAllReduce: {
+      type: Boolean,
+      value: false,
+    },
+    hasReplicaGroups: {
+      type: Boolean,
+      value: false,
+    },
+    isStepBreakdown: {
+      type: Boolean,
+      value: false,
+    },
+    stepBreakdownEle: {
+      type: Array,
+    },
+  },
+  /**
+   * Update the details card.
+   */
+  updateCard_: function(nodes) {
+    if (!nodes || nodes.length == 0) return;
+    this.isChannel = false;
+    this.isAllReduce = false;
+    this.isStepBreakdown = false;
+    this.hasReplicaGroups = false;
+    if (nodes[0].channelId) {
+      this.name = 'Channel #';
+      this.id = nodes[0].channelId;
+      this.isChannel = true;
+    } else if (nodes[0].hostName) {
+      this.name = 'Step breakdown of chip';
+      this.id = nodes[0].chipId;
+      this.isStepBreakdown = true;
+    } else if (nodes[0].replicaGroups) {
+      this.name = nodes[0].name;
+      this.id = null;
+      this.isAllReduce = true;
+      this.hasReplicaGroups = nodes[0].replicaGroups.length;
+    }
+  },
+  /**
+   * Converts from number of bytes to MiB.
+   */
+  bytesToMiB_: function(numBytes: number): number {
+    return numBytes / 1048576;
+  },
+  /**
+   * Return the formatted data size in MiB.
+   */
+  sizeMiB_: function(dataSize: undefined|number): string {
+    if (!dataSize) {
+      return '';
+    }
+    return this.format_(this.bytesToMiB_(dataSize));
+  },
+  /**
+   * Return the formatted link bandwidth in GiB/s.
+   * The link bandwidth here is defined by the data size transferred over the
+   * duration between the start of the send operation to the end of the
+   * recv-done operation.
+   */
+  bw_: function(dataSize: undefined|number, duration: undefined|number):
+      string {
+        if (!dataSize || !duration) {
+          return '';
+        }
+        return this.format_(dataSize / duration / 1073.74);
+      },
+  /**
+   * Return the chip id given the global core id.
+   */
+  chipId_: function(coreId: undefined|number): number {
+    if (!coreId) {
+      return 0;
+    }
+    return Math.floor(coreId / 2);
+  },
+  /**
+   * Return the node ordinal given the global core id.
+   */
+  nodeId_: function(coreId: undefined|number): number {
+    if (!coreId) {
+      return 0;
+    }
+    return coreId & 1;
+  },
+  /**
+   * Format a number with two digits after the decimal point.
+   */
+  format_: function(number: undefined|number): string {
+    return number == null ? '' : number.toFixed(2);
+  },
+  /**
+   * Return a formatted value associated with a specific breakdown.
+   */
+  getStepBreakdownValue_: function(node, key): string {
```

Comment:
What are the types for `node` and `key`? Here and below.

Reply:
Done, added StepBreakdownNode interface.